/*  Copyright 2012 MaidSafe.net limited

    This MaidSafe Software is licensed to you under (1) the MaidSafe.net Commercial License,
    version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
    licence you accepted on initial access to the Software (the "Licences").

    By contributing code to the MaidSafe Software, or to this project generally, you agree to be
    bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
    directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
    available at: http://www.maidsafe.net/licenses

    Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
    under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
    OF ANY KIND, either express or implied.

    See the Licences for the specific language governing permissions and limitations relating to
    use of the MaidSafe Software.                                                                 */

option optimize_for = LITE_RUNTIME;

package maidsafe.vault.protobuf;

message DataManagerValue {
  required uint64 size = 1;
  repeated bytes pmid_names = 2;
}

message DataManagerKeyValuePair {
  required bytes key = 1;
  required bytes value = 2;
}

message DataOrProof {
  message Data {
    required uint32 type = 1;
    required bytes name = 2;
    required bytes content = 3;
  }
  message Proof {
    required uint32 type = 1;
    required bytes name = 2;
    required bytes value = 3;
    required uint64 size = 4;
  }
  required bytes node_id = 1;
  optional bytes serialised_data = 2;
  required bytes serialised_proof = 3;
  required bytes signature = 4;
}

